### PR TITLE
Add support for recording to stream output

### PR DESCRIPTION
**Pull request by MiselKING**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature.

- **What is the current behavior?** (You can also link to an open issue here)
Currently screen recording only works for file output.

- **What is the new behavior (if this is a feature change)?**
Allow screen recording output to a stream.

- **Other information**:
It would be nice to allow stream output so we can stream to AWS for example.

### CODE REVIEW

**Review by prasanaworld — COMMENTED** [review]
It's mandatory to fix all the lint and UT cases before PR.

File: src/lib/pageVideoStreamWriter.ts [hunk 1]
```diff
@@ -24,14 +24,23 @@ export default class PageVideoStreamWriter extends EventEmitter {
...
-  constructor(savePath: string, options?: VideoOptions) {
+  constructor(savePath: string | Writable, options?: VideoOptions) {
     super();
 
     if (options) {
       this.options = options;
     }
 
-    this.configureVideoFile(savePath);
+    if (savePath && typeof savePath !== 'string') {
```
[the remark this answers is not in the document]
Comment (prasanaworld) [reply src/lib/pageVideoStreamWriter.ts]: Thanks @MiselKing 

PR looks fine to Me, But, The test case for the stream is failing, could u please fix it. 

Also, Shall I add your name to the contributor section? Please let me know.

**Comment by prasanaworld** [comment]
Thanks @MiselKING for raising the PR. Appreciate your help.

Since the build is failing. I picked you few changes of your PR and implemented along with multiple video format support.

I would like to add your name to contributor list. Please let me know, if your ok with this.

**Comment by MiselKING** [comment]
Hi @prasanaworld ,

Thank you for fixing this issue, I had a lot of work in these last few months so I never got the chance to tackle and fix this. 
Sure, I am ok with being the contributor, and also I will try and find more time to work on this library (by both answering questions and coding). Thanks.

**Pull request closed without merging**